crates/red_knot/src/workspace.rs

```diff
@@ -305,14 +309,18 @@ impl Package {
     }
 
     fn from_metadata(db: &dyn Db, metadata: PackageMetadata) -> Self {
-        Self::new(db, metadata.name, metadata.root, PackageFiles::default())
+        Self::builder(metadata.name, metadata.root, PackageFiles::default())
+            .durability(Durability::HIGH)
+            .new(db)
     }
 
     fn update(self, db: &mut dyn Db, metadata: PackageMetadata) {
         let root = self.root(db);
         assert_eq!(root, metadata.root());
 
-        self.set_name(db).to(metadata.name);
+        self.set_name(db)
+            .with_durability(Durability::HIGH)
```

Comment:
It's a bit annoying that Salsa downgrades the durability when setting a new value. I proposed changing the behavior in Salsa but I don't want to block this PR because of it.